homeassistant/components/media_player/cambridgeaudio.py

```diff
@@ -0,0 +1,366 @@
+    MediaPlayerDevice, PLATFORM_SCHEMA, MEDIA_TYPE_MUSIC)
+
+from homeassistant.const import (STATE_ON, STATE_OFF, STATE_PLAYING,
+                                 STATE_PAUSED, STATE_IDLE,
+                                 CONF_HOST, CONF_NAME, CONF_COMMAND_OFF)
+
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.dt as dt_util
+
+REQUIREMENTS = ['stream_magic==0.16']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default='Cambridge Audio Streamer'): cv.string,
+    vol.Optional(CONF_COMMAND_OFF, default='OFF'): cv.string
+})
+
+# supported features in all configurations
+SUPPORT_CAMBRIDGE = SUPPORT_TURN_OFF | SUPPORT_TURN_ON |\
+                    SUPPORT_PLAY | SUPPORT_SELECT_SOURCE |\
+                    SUPPORT_CLEAR_PLAYLIST | SUPPORT_STOP |\
+                    SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET |\
+                    SUPPORT_VOLUME_STEP | SUPPORT_SEEK | SUPPORT_PAUSE |\
+                    SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK |\
+                    SUPPORT_SHUFFLE_SET
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Cambridge Audio platform."""
+    from stream_magic import device as cadevice
+    from stream_magic import discovery as ca
+
+    hosts = []
+    sm = ca.StreamMagic()   # pylint: disable=C0103
+    # key to access known host list in hass.data
+    host_key = 'cambridgeaudio_hosts'
+
+    if host_key not in hass.data:
+        hass.data[host_key] = []
+
+    # found a new host that was specified in config file
+    if discovery_info is None:
+        host = config[CONF_HOST]
+        if host in hass.data[host_key]:
+            return
+
+        name = config[CONF_NAME]
+        poweroff_command = config[CONF_COMMAND_OFF]
+
+        try:
+            # discover() always returns a list, even for a single host
+            devices = sm.discover(host=host)
+            if devices:
+                dev = devices[0]
+                addr, port = dev[0][0:2]
+                desc = dev[1]['server']
+                scpd_url = dev[1]['location']
+                if not name:
+                    name = desc
+
+                smdevice = cadevice.StreamMagicDevice(addr, port,
+                                                      desc, scpd_url)
+                hosts.append(CADevice(smdevice, poweroff_command, name))
+                hass.data[host_key].append(host)
+                _LOGGER.debug("Added StreamMagic device with ip %s (%s)",
+                              addr, desc)
+        except OSError:
+            _LOGGER.debug("Unable to connect to device at %s", host)
+
+    else:   # netdisco found the device
+        host = discovery_info.get('host')
+        port = discovery_info.get('port')
+
+        # if a name was set in the config, use it
+        if CONF_NAME in config:
```

Comment:
The config will always be empty when setting up via discovery.

Reply:
Ok, I realized that now, too.
I thought I could override some settings (like the power off command) with the corresponding config options even when discovery is used.
I just tested it and you're right, that doesn't work.
With making CONF_HOST a required option, the config would fail on validation anyway.